models/src/main/scala/coop/rchain/models/HashM.scala

```diff
@@ -86,6 +86,14 @@ object HashM extends HashMDerivation {
         hashes <- value.toList.traverse(HashM[A].hash[F])
       } yield hashes.hashCode()
 
+  }
+  implicit def vectorHash[A: HashM]: HashM[Vector[A]] = new HashM[Vector[A]] {
+
+    override def hash[F[_]: Sync](value: Vector[A]): F[Int] =
+      for {
+        hashes <- value.toVector.traverse(HashM[A].hash[F])
+      } yield hashes.hashCode()
```

Comment:
`Vector` and `List` extends `Seq` so the new definition specific to Vector should not be needed. What is the reason to add these implementations for equals and hash code?

Reply:
Without them, the code will not compile; several errors will appear in those files. I think it is because in the definition for seq the type is explicity seq and does not use < to indicate that child classes can be used. I don't think Lists were being used by EqualsM and HashM, I think it was Seq that was being used so there were no conflicts with the old code.

Reply:
Makes sense. This will also ensure that no conversion is happening here. `value` is already and Vector so `.toVector` is redundant.

For comprehension is not needed, at the end it's just mapping `.map(_.hashCode)`.

Reply:
Changed!